src/driver/Flags.ml

```diff
@@ -0,0 +1,98 @@
+open Core
+
+type t =
+  { (* ------------------------------- *)
+    (* flags affecting code generation *)
+    optimization_level: Analysis_and_optimization.Optimize.optimization_level
+  ; allow_undefined: bool
+  ; functions_only: bool
+  ; standalone_functions: bool
+  ; use_opencl: bool
+  ; include_source: Frontend.Include_files.t
+        (* ------------------------- *)
+        (* flags which switch compiler "modes" *)
+  ; info: bool
+  ; version: bool
+  ; auto_format: bool
+  ; debug_settings: debug_settings
+        (* ------------------------- *)
+        (* flags for formatting mode *)
+  ; line_length: int
+  ; canonicalizer_settings: Frontend.Canonicalize.canonicalizer_settings
+        (* ------------------------- *)
+        (* extra settings *)
+  ; warn_pedantic: bool
+  ; warn_uninitialized: bool
+  ; filename_in_msg: string option }
+
+and debug_settings =
+  { debug_ast: bool
+  ; debug_typed_ast: bool
+  ; debug_mir: debug_options
```

Comment:
I don't like the "`debug_` prefix for every field" convention here. Maybe the prefixes could be more descriptive, like the ones that print intermediate representations could start with `print_`.